code/game/objects/items/tools/medical_tools.dm

```diff
@@ -0,0 +1,246 @@
+
+//CELLS here
+/obj/item/tool/geltool/gell  //Gell a CELL but GELL
+	name = "Gell classic vial"
+	desc = "A vial used to transport and store medical gel, NOT EDIBLE."
+	icon_state = "gell"
+	item_state = "gell"
+	icon = 'icons/obj/items/geltools.dmi'
+	var/gelcapacity = 840   //how much gel can you fit in this tool, THIS NUMBER WAS CHOSEN FOR ITS DIVISIBILITY
+	var/gelquantity = 840   //how much gel does it starts with
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/tool/geltool/gell/small
+	name = "Gell tiny vial"
+	desc = "A vial used to transport and store medical gel, NOT EDIBLE."
+	icon_state = "gelltiny"
+	item_state = "gelltiny"
+	gelcapacity = 280
+	gelquantity = 280
+	w_class = WEIGHT_CLASS_TINY
+
+
+/obj/item/tool/geltool/gell/attack(mob/living/M, mob/living/user)
+	if(M != user)
+		return ..()
+	if(gelquantity < 35)
+		return
+	user.visible_message("[user] sucks on \the [src].",
+	"You suck on \the [src].")
+	gelquantity -= 35 // 8 sucks should suffice them without killing them
+
+	playsound(loc, 'sound/voice/alien_drool2.ogg', 30, 1)
+	playsound(loc, 'sound/voice/skeleton_warcry.ogg', 10,1,1)
+	M.adjustFireLoss(10) // add a little bit of Burn damage
+	M.adjustToxLoss(5) // and tox for good measure
+	M.disabilities |= MUTE
+	addtimer(VARSET_CALLBACK(M, disabilities, M.disabilities & ~MUTE), 30 SECONDS)
+	to_chat(user, "<span class='warning'>You feel an acidic taste on you mouth as your lips tighten up!</span>")
+	update_icon()
+
+/obj/item/tool/geltool/gell/examine(mob/user)
+	..()
+	to_chat(user, "<b>The gel gauge measures: [gelquantity]/[gelcapacity]</b>")
+	update_icon()
+
+/obj/item/tool/geltool/gell/update_icon()
```

Comment:
override update_icon_state(), not update_icon

Reply:
above^